### PR TITLE
[api] Reinstall

### PR DESCRIPTION
**Pull request by mmalecki**
Previously a try to install a system which was already installed was
ignored. Now, `uninstall` script for the previously installed system is
run and new system is installed.

### CODE REVIEW

**Comment by indexzero** [comment]
@mmalecki It doesn't look like you're using semver here to compare if the version we're about to install is newer than the one which is currently installed. We should only uninstall if the current version does not satisfy the needs of the parent depending on it.

**Comment by mmalecki** [comment]
@indexzero That was my initial concept, but after discussion with @bmeck we decided to reinstall anyway.
Usecase: provisioning fails for some reason, I want to fix that. I just rerun the command on the machine.

**Comment by bmeck** [comment]
@indexzero can you clarify. My concern is if we push a breaking change like the following scenario (numbers made up):

Given the following install tree:

```
## Allows for revision level patches to be installed for haibu
haibu@1.0.0 ( deps: haibu-smartos: ~0.5.9 )
 | - haibu-smartos@0.5.10 # can be upgraded to haibu-smartos@0.5.11 or haibu-smartos@0.5.12
```

We update to `haibu-smartos@0.5.11` which includes a fix for `haibu-smartos@0.5.10`

If we don't allow a uninstall if the error is non-trivial to detect when installing. The logic must be included in any future versions which satisfy haibu's haibu-smartos dependency version.

So. In this case we could detect an installed error in `haibu-smartos@0.5.11`, however, if we use error detection instead of uninstalls it must always be included for any version past `haibu-smartos@0.5.11` that may be upgraded to from a defunct installation.

In other words:

If `haibu-smartos@0.5.11` must detect an error in `haibu-smartos@0.5.10`, and if `haibu-smartos@0.5.10` may be fast-forwarded to `haibu-smartos@0.5.12`, then `haibu-smartos@0.5.12` must include the error detection done in `haibu-smartos@0.5.11`

I would be fine with keeping this behind a flag, but we should have a way to uninstall when upgrading easily. I was going to think of that as the default behavior, but we could talk about it more if you don't want to uninstall things.

@indexzero if we don't want uninstalls to be default, how does a flag for --clean-install sound?

**Comment by mmalecki** [comment]
It is now only enabled when called with `--force`/`-f` flag.

**Comment by bmeck** [comment]
lgtm

**Comment by mmalecki** [comment]
Merged.
